src/messages/mod.rs

```diff
@@ -78,6 +80,32 @@ impl HopMessage {
     }
 }
 
+/// Metadata needed for verification of the sender
+#[derive(Ord, PartialOrd, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
+pub struct SecurityMetadata {
+    proof: SectionProofChain,
+    sender_prefix: Prefix<XorName>,
+    signature: BlsSignature,
+}
+
+impl SecurityMetadata {
+    pub fn verify_sig(&self, bytes: Vec<u8>) -> bool {
```

Comment:
so the `bytes` is the signature?
It seems it has to be the signature of the last public key?

Reply:
`bytes` is the serialised message. The `SecurityMetadata` itself has no access to the message it is attached to, so the actual message needs to be passed as an argument.